app/src/org/commcare/activities/connectId/fragments/PhoneFragment.java

```diff
@@ -0,0 +1,315 @@
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for getContext() fragment
+        String existing;
+        View view = inflater.inflate(R.layout.screen_connect_primary_phone, container, false);
+        titleTextView = view.findViewById(R.id.connect_primary_phone_title);
+        messageTextView = view.findViewById(R.id.connect_primary_phone_message);
+        countryCodeInput = view.findViewById(R.id.connect_primary_phone_country_input);
+        phoneInput = view.findViewById(R.id.connect_primary_phone_input);
+        availabilityTextView = view.findViewById(R.id.connect_primary_phone_availability);
+        button = view.findViewById(R.id.connect_primary_phone_button);
+        countryCodeInput.addTextChangedListener(watcher);
+        phoneInput.addTextChangedListener(watcher);
+        button.setOnClickListener(v -> handleButtonPress());
+        requireActivity().setTitle(getString(R.string.connect_phone_page_title));
+//        method = getIntent().getStringExtra(ConnectConstants.METHOD);
+        //Special case for initial reg. screen. Remembering phone number before account has been created
+//        existingPhone = getIntent().getStringExtra(ConnectConstants.PHONE);
+        ConnectUserRecord user = ConnectManager.getUser(getActivity());
+        String title = getString(R.string.connect_phone_title_primary);
+        String message = getString(R.string.connect_phone_message_primary);
+
+        if (!method.equals(ConnectConstants.METHOD_CHANGE_ALTERNATE))
+            PhoneNumberHelper.requestPhoneNumberHint(getActivity());
+
+        if (method.equals(ConnectConstants.METHOD_CHANGE_ALTERNATE)) {
+            title = getString(R.string.connect_phone_title_alternate);
+            message = getString(R.string.connect_phone_message_alternate);
+
+            existing = user != null ? user.getAlternatePhone() : null;
+        } else {
+            existing = user != null ? user.getPrimaryPhone() : existingPhone;
+        }
+
+        titleTextView.setText(title);
+        messageTextView.setText(message);
+        displayNumber(existing);
+
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        checkPhoneNumber();
+
+        KeyboardHelper.showKeyboardOnInput(requireActivity(), phoneInput);
+
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        String phone = PhoneNumberHelper.handlePhoneNumberPickerResult(requestCode, resultCode, data, getActivity());
+        skipPhoneNumberCheck = false;
+        displayNumber(phone);
+    }
+
+    public void finish(boolean success, String phone) {
+//        Intent intent = new Intent(getIntent());
```

Comment:
Similar to data being passed in, we need a way to indicate that this page is done and pass any data on to the next step. Eventually this will be done via Navigation/SafeArgs. For now, maybe the caller should pass in a callback function so it can handle the next step?

This may require more thought...